pulp_container/app/authorization.py

```diff
@@ -151,6 +147,8 @@ def has_pull_permissions(self, path):
             distribution = ContainerDistribution.objects.get(base_path=path)
         except ContainerDistribution.DoesNotExist:
             return False
+        if not distribution.private:
+            return True
```

Comment:
This is the real meat in this PR.

Reply:
i believe this check should happen couple of lines above.
If distribution is not marked as private, there is no need to check on any permissions pull permissions https://github.com/pulp/pulp_container/pull/205/files#diff-c9c63122c0e9c033b596c104536853afe2505833c6d33f9ed1e06bf7322ef19fR143

Reply:
This is the first line after having a grasp on the distribution object. Do you mean to move that too?

Reply:
I thought it was convenient to see this flag being evaluated as part of the question for pull permissions.
Being rather commonly set to True, can rearrange that a bit however.

Reply:
In the long run, i see `has_pull_permission` being transformed into an access policy.

Reply:
i think even in the short run. AH plan to have their own policy, i moved this story to mvp milestone https://pulp.plan.io/issues/8075

Reply:
seems like after rebase the code looks different. we should check distribution.private flag in the first place before checking for any perms.. i know this will be re-worked again in other prs but let's keep the logic in order in every pr, wdyt?